### PR TITLE
Closes: #364 Inconsistency when searching files on dataset page

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetFilesTab.java [hunk 1]
```diff
@@ -342,13 +341,24 @@ public void refreshPaginator() {
         rowsPerPage = dt.getRowsToRender();
     }
 
+    /**
+     * Reset page number to the first element found for showing search results.
+     */
+    public void resetPaginator() {
+        RequestContext.getCurrentInstance().execute("PF('filesTable').getPaginator().setPage(0)");
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetFilesTab.java]: I think we should keep this kind of code in frontend.

File: dataverse-webapp/src/main/webapp/filesFragment.xhtml [hunk 2]
```diff
@@ -97,23 +111,31 @@
...
                 <div class="col-xs-12 bg-warning text-left" style="font-weight:normal;padding-top:.5em;">
                     <!-- SELECTION MESSAGE -->
                     <p>
-                        <h:outputFormat value="#{bundle['file.numFilesSelected']}">
+                        <h:outputFormat rendered="#{datasetFilesTab.selectedFiles.size() gt 0}"
+                                        value="#{bundle['file.numFilesSelected']}">
                             <f:param value="#{datasetFilesTab.selectedFiles.size()}"/>
                         </h:outputFormat>
-                        <ui:fragment rendered="#{datasetFilesTab.selectedFiles.size() lt datasetFilesTab.workingVersion.fileMetadatas.size()}">
+                        <ui:fragment rendered="#{datasetFilesTab.selectedFiles.size() lt
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/filesFragment.xhtml]: Imo complex boolean logic should be move to backend with descriptive function name, this is kinda hard to read.

File: dataverse-persistence/src/main/resources/Bundle_en.properties [hunk 3]
```diff
@@ -1427,7 +1427,8 @@ file.count={0} to {1} of {2} {2, choice, 0#Files|1#File|2#Files}
 file.count.shown={0} {0, choice, 0#Files Selected|1#File|2#Files}
 file.clearSelection=Clear selection.
 file.numFilesSelected={0} {0, choice, 0#files are|1#file is|2#files are} currently selected.
-file.selectAllFiles=Select all {0} files in this dataset.
+file.selectAllFiles=Select all {0} files in this dataset{1}
+file.selectAllFilesWhileSearching=\u0020matching search term.
```
Comment (MrK191) [inline dataverse-persistence/src/main/resources/Bundle_en.properties]: Hmm imo bundle text should be standalone (they don't need another bundle to make sense). To be discussed during daily.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetFilesTab.java (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetFilesTab.java]: Actually i couldn't make it work in frontend but I'll give it another try.

File: dataverse-webapp/src/main/webapp/filesFragment.xhtml (hunk 2 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/filesFragment.xhtml]: This crossed my mind aswell. Could we discuss this on daily maybe?

**Review by MrK191 — APPROVED** [review]

**Pull request merged**